### PR TITLE
View by session for My courses tab - version 1.9.10.2

### PR DESCRIPTION
**Pull request by baelmyhu**
Add view by session link in My courses tab

Add line in configuration.dist.php
// For view by session in the My courses tab
$_configuration['my_courses_view_by_session'] = false;

### CODE REVIEW

**Comment by jmontoyaa** [comment]
I can't accept this PR my previous comments were not applied see:

https://github.com/chamilo/chamilo-lms/pull/631
https://github.com/chamilo/chamilo-lms/pull/629

File: main/inc/lib/course.lib.php [hunk 1]
```diff
@@ -3906,17 +3906,152 @@ public static function display_courses_in_category($user_category_id, $load_dirs
...
-     * @author Patrick Cool <patrick.cool@UGent.be>, Ghent University
+     * @param string $userId
      * @return array containing all the titles of the user defined courses with the id as key of the array
      */
-    public static function get_user_course_categories()
+    public static function get_user_course_categories($userId = "")
     {
         global $_user;
+        if ($userId == "") {
+            $RealUserId = $_user['user_id'];
```
Comment (jmontoyaa) [inline main/inc/lib/course.lib.php]: Use cameCase $realUserId and not $RealUserId

File: main/inc/lib/course.lib.php [hunk 2]
```diff
@@ -3906,17 +3906,152 @@ public static function display_courses_in_category($user_category_id, $load_dirs
...
     /**
      * Retrieves the user defined course categories
-     * @author Patrick Cool <patrick.cool@UGent.be>, Ghent University
+     * @param string $userId
      * @return array containing all the titles of the user defined courses with the id as key of the array
      */
-    public static function get_user_course_categories()
+    public static function get_user_course_categories($userId = "")
     {
         global $_user;
```
Comment (jmontoyaa) [inline main/inc/lib/course.lib.php]: Don't use globa $_user you can use $user = api_get_user_info()

**Comment by jmontoyaa** [comment]
This change should be optional. It seems that is going to work with out the possibility of turn it on/off?

File: main/inc/lib/course.lib.php (hunk 2 above)
Comment (baelmyhu) [inline main/inc/lib/course.lib.php]: $_user was use in this function before I've modified it, I correct it and send you a new PR, sorry for the time you spend on it :sa:

File: main/inc/lib/course.lib.php (hunk 1 above)
Comment (baelmyhu) [inline main/inc/lib/course.lib.php]: ooops, a last mistake, I correct it

File: main/inc/lib/sessionmanager.lib.php [hunk 3]
```diff
@@ -5543,4 +5538,405 @@ public static function getTotalUserCoursesInSession($sessionId)
...
+        return false;
+    }
+
+
+    /**
+     * Return database info for session with id $inSessionId
+     * @param $sessionId
+     * @return array
+     */
+    public function get_session_info_by_id($sessionId)
```
Comment (jmontoyaa) [inline main/inc/lib/sessionmanager.lib.php]: You can use api_get_session_info()

**Pull request closed without merging**